var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -167,6 +189,30 @@ def cmake_args(self):
                     '-DModule_vtkGUISupportQtOpenGL:BOOL=ON',
                 ])
 
+        if '+xdmf' in spec:
+            cmake_args.extend([
+                # Enable XDMF Support here
+                "-DCMAKE_POLICY_DEFAULT_CMP0074=NEW",
+                "-DModule_vtkIOXdmf2:BOOL=ON",
+                "-DModule_vtkIOXdmf3:BOOL=ON",
+                "-DModule_vtkIOParallelXdmf3:BOOL=ON",
+                "-DBOOST_ROOT={0}".format(spec['boost'].prefix),
+                "-DBOOST_LIBRARY_DIR={0}".format(spec['boost'].prefix.lib),
+                "-DBOOST_INCLUDE_DIR={0}".format(spec['boost'].prefix.include),
+                "-DBOOST_NO_SYSTEM_PATHS:BOOL=ON",
+                # This is needed because VTK has multiple FindBoost
+                # and they stick to system boost if there's a system boost
+                # installed with CMake
+                "-DBoost_NO_BOOST_CMAKE:BOOL=ON",
+                "-DHDF5_ROOT={0}".format(spec['hdf5'].prefix),
+                # The xdmf project does not export any CMake file...
+                "-DVTK_USE_SYSTEM_XDMF3:BOOL=OFF",
```

Comment:
If you want, you could also add a `xmdf2` and `xmdf3` package to spack and use that instead of the vendored versions.

Reply:
I tried that path... The problem is that the `xdmf` git repo does not ship with a `xdmf3-config.cmake`, so when you enable `VTK_USE_SYSTEM_XDMF{2,3}`, the `FindXdmf{2,3}` does not find the cmake config files... 

To overcome this I believe we should ship upstream a patch to provide a `-config.cmake` or patching the `VTK` to use a smarter `FindXdfm3`. 

I think it's easier to leverage the vendored versions...